plugins/inputs/logfile/logfile_test.go

```diff
@@ -1063,3 +1063,22 @@ func TestGenerateLogGroupName(t *testing.T) {
 		logGroupName,
 		expectLogGroup))
 }
+
+func TestCheckForDuplicateRetentionSettingsPanics(t *testing.T) {
+	tt := NewLogFile()
+	logGroupName := "DuplicateLogGroupName"
+	tt.FileConfig = []FileConfig{{
+		FilePath:        "SampleFilePath",
+		FromBeginning:   true,
+		LogGroupName:    logGroupName,
+		RetentionInDays: 1,
+	},
+		{
+			FilePath:        "SampleFilePath",
```

Comment:
One thing I just thought about - do we have to account for case sensitivity for file paths? IIRC Linux/Unix are case sensitive but Windows is case insensitive. Given that, I think it's possible for this to not work on Windows. We build a map of log group names to count their occurrences, but I _know_ that the internal map keys are case sensitive.

Reply:
This is a good consideration specifically for log group names. I changed it to normalize strings as lowercase for logGroupNames.